gossipsub.go

```diff
@@ -304,19 +304,23 @@ func (gs *GossipSubRouter) pxConnect(peers []*pb.PeerInfo) {
 			continue
 		}
 
-		var srr *routing.SignedRoutingState
-		var err error
-		if pi.SignedAddrs != nil {
+		var srr *record.Envelope
+		if pi.SignedPeerRecord != nil {
 			// the peer sent us a signed record; ensure that it is valid
-			srr, err = routing.UnmarshalSignedRoutingState(pi.SignedAddrs)
+			envelope, r, err := record.ConsumeEnvelope(pi.SignedPeerRecord, peer.PeerRecordEnvelopeDomain)
 			if err != nil {
 				log.Warningf("error unmarshalling routing record obtained through px: %s", err)
 				continue
 			}
-			if srr.PeerID != p {
-				log.Warningf("bogus routing record obtained through px: peer ID %s doesn't match expected peer %s", srr.PeerID, p)
+			rec, ok := r.(*peer.PeerRecord)
+			if !ok {
+				log.Warnf("bogus routing record obtained through px: envelope payload is not PeerRecord")
+			}
+			if rec.PeerID != p {
```

Comment:
there is a control flow issue here; if `!ok` then `rec.PeerID` is a nil pointer dereference, which could crash.

Reply:
thanks for catching that 👍